WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt

```diff
@@ -113,15 +120,20 @@ class ReaderDiscoverViewModel @Inject constructor(
         // Listen to changes to the discover feed
         _uiState.addSource(readerDiscoverDataProvider.discoverFeed) { posts ->
             launch {
-                if (posts != null && posts.cards.isNotEmpty()) {
-                    val discoverFeedContainsOnlyWelcomeCard = posts.cards.size == 1 &&
-                            posts.cards.filterIsInstance<WelcomeBannerCard>().isNotEmpty()
-                    if (!discoverFeedContainsOnlyWelcomeCard) {
-                        _uiState.value = ContentUiState(
-                                convertCardsToUiStates(posts),
-                                reloadProgressVisibility = false,
-                                loadMoreProgressVisibility = false
-                        )
+                val userTags = getFollowedTagsUseCase.get()
+                if (userTags.isEmpty()) {
+                    _uiState.value = ShowFollowInterestsEmptyUiState
+                } else {
+                    if (posts != null && posts.cards.isNotEmpty()) {
```

Comment:
@malinajirka - Is there ever a case in which the posts can be null or empty? In the future should we think about adding a generic "empty" view ? Not related to this PR, but while I was testing I forced empty data and was shown a blank screen.

Reply:
AFAIK it can't happen. I've noticed this potential issue as well. I think showing an empty view with "refresh" button sounds like a good solution just to make sure.